### PR TITLE
Change table organism to be in line with design manual

### PR DESCRIPTION
**Pull request by contolini**
@nataliafitzgerald  noticed that our table organism had outdated styles for smaller screens. I moved the new styles into our tables stylesheet and updated all the classes from `table__stack-on-small` to `o-table__stack-on-small` which is what CF v4 uses.

## Changes

- Updated the table organism to include the latest design manual stacking styles.

## Screenshots

Before:

screen shot 2016-12-05 at 5 11 53 pm

After:

screen shot 2016-12-05 at 4 49 02 pm

## Review

- @cfpb/cfgov-frontends

### CODE REVIEW

File: cfgov/unprocessed/css/enhancements/tables.less [hunk 1]
```diff
@@ -120,44 +120,40 @@
         }
     }
 
-    .respond-to-max(960px, {
+}
 
-        font-size: 14px;
+.respond-to-max( @bp-xs-max, {
+    .o-table {
+        width: 100%;
```
Comment (sebworks) [inline cfgov/unprocessed/css/enhancements/tables.less]: Is this style necessary?

**Comment by sebworks** [comment]
Looks good, I would check with @jimmynotjim to verify if any of the styles removed are necessary on legacy blog tables.

File: cfgov/unprocessed/css/enhancements/tables.less (hunk 1 above)
Comment (contolini) [inline cfgov/unprocessed/css/enhancements/tables.less]: Nope! Good call -- it's covered by the rule above it.

**Comment by schaferjh** [comment]
Cool beans. Does this also apply to our tables that currently losing headings at mobile? Examples:
http://www.consumerfinance.gov/policy-compliance/rulemaking/small-business-review-panels/
http://www.consumerfinance.gov/policy-compliance/guidance/implementation-guidance/title-xiv-mortgage-rules/
http://www.consumerfinance.gov/data-research/credit-card-data/college-credit-card-agreement/

Would be nice to get those in line as well -- or are they a different beast?

**Comment by contolini** [comment]
@schaferjh Yep!

Before:

screen shot 2016-12-06 at 11 06 52 pm

After:

screen shot 2016-12-06 at 11 06 12 pm

(Links in tables are still a bit awkward.)

**Comment by schaferjh** [comment]
👍

File: cfgov/unprocessed/css/enhancements/tables.less [hunk 2]
```diff
@@ -120,44 +120,36 @@
...
+    .o-table__stack-on-small {
+        tr,
+        td {
+            display: block;
         }
 
-        tr {
-            border-bottom: none;
+        & > thead {
+          display: none
```
Comment (jimmynotjim) [inline cfgov/unprocessed/css/enhancements/tables.less]: Should be four space indentation to match

File: cfgov/unprocessed/css/enhancements/tables.less [hunk 3]
```diff
@@ -120,44 +120,36 @@
...
+.respond-to-max( @bp-xs-max, {
+    .o-table__stack-on-small {
+        tr,
+        td {
+            display: block;
         }
 
-        tr {
-            border-bottom: none;
+        & > thead {
```
Comment (jimmynotjim) [inline cfgov/unprocessed/css/enhancements/tables.less]: Good call restricting this.

File: cfgov/unprocessed/css/enhancements/tables.less [hunk 4]
```diff
@@ -120,44 +120,36 @@
...
-            .h4();
-            margin-bottom: 10px;
-            background-color: @td-bg-mobile;
+        td[data-label]:before {
+            .webfont-demi();
+            display: block;
+            margin-top: 0;
+            margin-bottom: .41666667em;
+            content: attr(data-label);
+            font-size: .875em;
```
Comment (jimmynotjim) [inline cfgov/unprocessed/css/enhancements/tables.less]: Update this to `unit( @size-v / @base-font-size-px, em );` so it's clear where it's coming from and will update automatically if that value ever changes.

File: cfgov/unprocessed/css/enhancements/tables.less [hunk 5]
```diff
@@ -120,44 +120,36 @@
...
 
-        td:first-child {
-            .h4();
-            margin-bottom: 10px;
-            background-color: @td-bg-mobile;
+        td[data-label]:before {
+            .webfont-demi();
+            display: block;
+            margin-top: 0;
+            margin-bottom: .41666667em;
```
Comment (jimmynotjim) [inline cfgov/unprocessed/css/enhancements/tables.less]: Update this to use a unit calculation so it's clear what the final computed value should be.

File: cfgov/unprocessed/css/enhancements/tables.less [hunk 6]
```diff
@@ -120,44 +120,36 @@
...
         }
 
-        td:first-child {
-            .h4();
-            margin-bottom: 10px;
-            background-color: @td-bg-mobile;
+        td[data-label]:before {
+            .webfont-demi();
+            display: block;
+            margin-top: 0;
```
Comment (jimmynotjim) [inline cfgov/unprocessed/css/enhancements/tables.less]: Is this reset necessary? I can't find where it's set in the code but I'm not sure if it's in our reset or a browser default.

File: cfgov/unprocessed/css/enhancements/tables.less [hunk 7]
```diff
@@ -120,44 +120,36 @@
...
+            margin-bottom: .41666667em;
+            content: attr(data-label);
```
Comment (jimmynotjim) [inline cfgov/unprocessed/css/enhancements/tables.less]: Minor: Add spaces within parens.

**Comment by jimmynotjim** [comment]
> Looks good, I would check with @jimmynotjim to verify if any of the styles removed are necessary on legacy blog tables.

I didn't think we updated the blog tables to use the small screen styles since that would have required adding the class. @KimberlyMunoz might remember better than me though.

**Comment by sebworks** [comment]
💯, after comments.

**Pull request merged**